crates/client/db/src/lib.rs

```diff
@@ -249,6 +245,19 @@ impl DatabaseExt for DB {
     }
 }
 
+#[derive(Debug)]
+pub struct TrieLogConfig {
+    pub max_saved_trie_logs: usize,
+    pub max_saved_snapshots: usize,
+    pub snapshot_interval: u64,
+}
+
+impl Default for TrieLogConfig {
+    fn default() -> Self {
+        Self { max_saved_trie_logs: 0, max_saved_snapshots: 0, snapshot_interval: 5 }
+    }
```

Comment:
Does it make sense to have `max_saved_snapshots: 0` but `snapshot_interval: 5`?

Reply:
no but it allows this
```rs
TrieLogConfig { max_saved_trie_logs: 50, max_saved_snapshots: 10, ..Default::default() }
```


i don't like these 3 arguments but that's how the bonsai trie works right now and i'd love to spend the time to redo all of this